src/api/upload.rs

```diff
@@ -7,16 +7,18 @@ use failure::Error;
 use libnest::package::NPFExplorer;
 use rocket::http::Status;
 use rocket::{Data, State};
+use rocket_contrib::json;
+use rocket_contrib::json::{JsonValue};
```

Comment:
How about `use rocket_contrib::json::{self, JsonValue};` :p

Reply:
You'd be surprise to discover that... it doesn't work :D  (I hoped you would ask that :D)

Here `rocket_contrib::json` is a macro, not a module, so using `rocket_contrib::json::self` wouldn't refer to the same item.

(Yes, it sucks that Rust allows two items in the same scope with the same name in the first place, I agree, but here we are)

Reply:
I think "Bleh :nauseated_face:" is the appropriate reaction to this